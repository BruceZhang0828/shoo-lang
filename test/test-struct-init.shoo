struct FunStuff {
	string hello;
	int five;
}
FunStuff hi;

function test_if_malloc() int {
	hi = {hello = "hello"; five = 5;};
	return 5;
}

test_if_malloc();
if (hi.five == 5){
	println("MALLOC DOES WORK");
}

println(str_of_int(hi.five));
println(hi.hello);
