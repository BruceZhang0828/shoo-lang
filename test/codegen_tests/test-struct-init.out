MALLOC DOES WORK
5
hello